cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -326,6 +335,38 @@ impl YaccParser {
         }
     }
 
+    fn parse_action(&mut self, i: usize) -> YaccResult<(usize, String)> {
+        let mut j = i;
+        let mut c = 0; // Count braces
+        while j < self.src.len() {
+            let ch = self.src[j..].chars().next().unwrap();
+            c += match ch {
+                '{' => 1,
+                '}' => -1,
+                _ => 0
+            };
+            if c == 0 {
```

Comment:
Should we check this in the `}` clause above, since neither of the other two clauses can trigger this (I think)?

Reply:
Can't the `_` clause trigger this?

Reply:
`parse_action` is only called if the input starts with a `{`:

```rust
            } else if self.lookahead_is("{", i).is_some() {
                let (j, a) = self.parse_action(i)?;
```

so, by definition, the first time this loop executes, c will be set to 1, and thus the `_` clause can't cause can't trigger this.

Reply:
You are right. Now I'm just trying to decide if this is more readable:
```
c += match ch {
    '{' => 1,
    '}' => {
        if c == 1 { break };
        -1
    }
    _ => 0
};
```

Reply:
I think you can do:

```rust
c += match ch {
    '{' => 1,
    '}' if c == 1 => { break; }
    '}' => -1
    _ => 0
};
```

Reply:
Did this make it in yet?

Reply:
I would have to write this as
```
c += match ch {
    '{' => 1,
    '}' if c == 1 => {
        c = 0;
        break;
    }
    '}' => -1
    _ => 0
};
```

So I am still uncertain if this increases readability.

Reply:
I think either of the solutions you've suggested here are OK: but they are better than what's currently in the commit, as they make clear that the condition is satisfied within the loop.

Reply:
Okay, I've updated it.